app/uk/gov/hmrc/gform/eval/EvaluationResults.scala

```diff
@@ -234,13 +205,8 @@ case class EvaluationResults(
         if (booleanExprResolver.resolve(cond)) loop(field1) else loop(field2)
       case Else(field1: Expr, field2: Expr) => loop(field1) orElse loop(field2)
       case ctx @ FormCtx(formComponentId)   => get(ctx, fromVariadicValue, evaluationContext)
-      case Sum(FormCtx(formComponentId))    => calculateSum(formComponentId, recData, unsupportedOperation("Number")(expr))
-      case Sum(field1) =>
-        loop(field1) match {
-          case lrs: ListResult =>
-            lrs.list.fold(NumberResult(0)) { case (a, b) => a + b }
-          case _ => unsupportedOperation("Number")(expr)
-        }
+      case Sum(_) =>
+        loop(expr.mapExpr(sumUtil.f))
```

Comment:
Would it be possible to reimplement `mapExpr` so it follows pattern like https://github.com/hmrc/gform/blob/main/app/uk/gov/hmrc/gform/formtemplate/Substituter.scala 
https://github.com/hmrc/gform/blob/main/app/uk/gov/hmrc/gform/formtemplate/Substituter.scala ?
It would be nice if all implementation was in single file.